### ISSUE TITLE
Make pickup patches optional

### ISSUE THREAD

**Issue by MayberryZoom** [randovania/open-dread-rando#374]
`pickups` should be optional in the JSON, so ODR can easily be used to make ROM hacks that leave items in their vanilla locations. I'm making this issue to discuss the viability of such a change and how much work it would take if pursued.

My assumption is that the main concern is the new pickups? Maybe it should only add pickups to bosses if they're included in the `pickups` field?

I'm also not sure if this would cause issues with split pickups. In theory things like Plasma Beam could be left unmodified on the statue, but I'm not sure if that would play nice with the rest of the code or work well in-game.

A related note: currently, pickups is required to modify every pickup location. If we can get the code to play nice with optional `pickups`, then we can also allow JSONs that only modify a small handful of pickups. This would be in line with how every other field works in the JSON.

**Comment by duncathan** [comment]
our pickups have significant enough differences from vanilla pickups that this would be pretty hard to accomplish. you've identified a couple of the issues 
- custom pickups (e.g. bosses) would at least need a close looking at
- split pickups absolutely will break
- even worse, allowing e.g. vanilla plasma AND an ODR wide or wave would break catastrophically. this makes editing only some pickups even less viable than editing none at all
- you'd need to double check that vanilla post-pickup stuff hasn't been broken by other changes we make

**Comment by MayberryZoom** [comment]
Yeah I figured this would be difficult to accomplish. I think this is definitely a long term goal and something we should try to fit into an eventual refactor of the pickups system. It seems infeasible with the current state of things, but I think there's other desired features/changes that would need a refactor or partial refactor anyways.

**Comment by duncathan** [comment]
to be clear: making the `pickups` field optional is viable but will require a lot of work. allowing a `pickups` field with only a small handful of pickups will never be viable. it's all or nothing

**Comment by MayberryZoom** [comment]
Could we just replace the vanilla beam locations with ODR beams if they're not provided? That seems like a better solution to me.
